tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py

```diff
@@ -21,48 +17,57 @@
 class TimePicker(Base):
 
     _time_picker_locator = (By.CLASS_NAME, 'value-selector-time-picker')
-    _done_button_locator = (By.CSS_SELECTOR, '.value-selector-time-picker-buttons > button.value-selector-confirm')
-    _hour_picker_locator = (By.CLASS_NAME, 'value-picker-hours')
-    _minutes_picker_locator = (By.CLASS_NAME, 'value-picker-minutes')
-    _hour24_picker_locator = (By.CLASS_NAME, 'value-picker-hour24-state')
+    _done_button_locator = (By.CSS_SELECTOR,
+                            '.value-selector-time-picker-buttons > button.value-selector-confirm')
+
+    _hour_picker_locator = (By.CSS_SELECTOR, '.value-picker-hours > div')
+    _minutes_picker_locator = (By.CSS_SELECTOR, '.value-picker-minutes > div')
+    _hour24_picker_locator = (By.CSS_SELECTOR, '.value-picker-hour24-state > div')
 
     def __init__(self, marionette):
         Base.__init__(self, marionette)
         self.marionette.switch_to_frame()
-        Wait(self.marionette).until(expected.element_displayed(*self._hour_picker_locator))
+        self.wait_for_element_displayed(*self._current_element(*self._hour_picker_locator))
 
-        # TODO: wait for the time picker to fade in Bug 1038186
+        # TODO: wait for the time piker to fade in Bug 1038186
         time.sleep(2)
 
     def tap_done(self):
-        time_picker = self.marionette.find_element(*self._time_picker_locator)
         self.marionette.find_element(*self._done_button_locator).tap()
-        Wait(self.marionette).until(expected.element_not_displayed(time_picker))
-        # TODO: wait for the time picker to fade out Bug 1038186
+        self.wait_for_element_not_displayed(*self._time_picker_locator)
```

Comment:
ah, i was pasting over the current code in master because I reverted the change.  I'll change it to Wait() call.